modal/_resources.py

```diff
@@ -9,13 +9,28 @@
 
 def convert_fn_config_to_resources_config(
     *,
-    cpu: Optional[float],
+    cpu: Optional[Union[float, Tuple[float, float]]],
     memory: Optional[Union[int, Tuple[int, int]]],
     gpu: GPU_T,
     ephemeral_disk: Optional[int],
 ) -> api_pb2.Resources:
     gpu_config = parse_gpu_config(gpu)
-    milli_cpu = int(1000 * cpu) if cpu is not None else None
+    if cpu and isinstance(cpu, tuple):
+        if not cpu[0]:
+            raise InvalidError("CPU request in tuple must not be None")
+        elif not cpu[1]:
+            raise InvalidError("CPU limit in tuple must not be None")
```

Comment:
Tiny nit, but this error would be confusing if either are passed as `0`.

Reply:
Hmm yeh not a nit in my opinion. It's bad.